tests/system/test_dbapi.py

```diff
@@ -810,3 +805,33 @@ def test_dml_returning_delete(shared_instance, dbapi_database, autocommit):
     assert cur.fetchone() == (1, "first-name")
     assert cur.rowcount == 1
     conn.commit()
+
+
+def _execute_common_precommit_statements(cursor: Cursor):
+    # execute several DML statements within one transaction
+    cursor.execute(
+        """
+        INSERT INTO contacts (contact_id, first_name, last_name, email)
```

Comment:
I forget, does this client support Spanner's generic batch DML API yet?  If so, should these three statements be batched?  It'd be a significant performance win, aside from testing the batch-DML API.

(I'm thinking a bit about test performance:  Python is not a fast language.  In my experience, Python test suites have a tendency to grow and become slower over time to the point where it becomes annoying to wait on the test suite.  So I personally like to put at least a little thought into test performance up front, in an attempt to mitigate this.)

Reply:
Cursor has executemany() method which supports batch DML in a limited way where one can only pass single sql query and a list of parameters to run the query with.
I had discussed this with Knut and we have a plan to implement a method in Cursor where multiple sql queries can be passed and run as a batch DML

Reply:
Makes sense.

Sounds like that method doesn't exist yet?  Maybe it would make sense to update this code someday in the future; no need to change it now, though.

Reply:
Yes it doesn't exist yet. Created b/312625672 for tracking